tests/framework/extensions/provisioning/ssh.go

```diff
@@ -8,38 +8,75 @@ import (
 	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
-	cpuUsageVar = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
-	checkCPU    = "CheckCPU"
+	cpuUsageVar                                   = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
+	checkCPU        provisioninginput.SshTestCase = "CheckCPU"
+	checkCPUCommand                               = "ps -C agent -o %cpu --no-header"
+	nodeReboot      provisioninginput.SshTestCase = "NodeReboot"
+	activeState                                   = "active"
 )
 
 // This func tests the ssh tests specified in the provisioninginput config clusterSSHTests field.
 // For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
-func CallSSHTestByName(testName string, node *nodes.Node) error {
-	switch testName {
+func CallSSHTestByName(testCase provisioninginput.SshTestCase, node *nodes.Node, client *rancher.Client, clusterID string) error {
+	switch testCase {
 	case checkCPU:
-		command := "ps -C agent -o %cpu --no-header"
-		output, err := node.ExecuteCommand(command)
+		output, err := node.ExecuteCommand(checkCPUCommand)
 		if err != nil {
 			return err
 		}
-		str_output := output[:strings.IndexByte(output, '\n')]
-		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + str_output + "%")
+		strOutput := output[:strings.IndexByte(output, '\n')]
+		logrus.Infof("CheckCPU test on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, strOutput)
 
-		output_int, err := strconv.ParseFloat(strings.TrimSpace(str_output), 32)
-		if output_int > cpuUsageVar {
-			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + str_output + "%")
+		outputInt, err := strconv.ParseFloat(strings.TrimSpace(strOutput), 32)
+		if outputInt > cpuUsageVar {
+			logrus.Warnf("Cluster agent cpu usage is too high on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, strOutput)
 		}
 		if err != nil {
 			return err
 		}
+	case nodeReboot:
+		logrus.Infof("Running NodeReboot test on node %s", node.PublicIPAddress)
+		command := "sudo reboot"
+		_, err := node.ExecuteCommand(command)
+		if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
+			return err
+		}
+		time.Sleep(10 * time.Second)
+		err = nodev1.AllMachineReady(client, clusterID)
+		if err != nil {
+			logrus.Errorf("Node %s failed to reboot successfully", node.PublicIPAddress)
+			return err
+		}
+
+		err = wait.Poll(1*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
Do this instead: https://github.com/rancher/rancher/blob/385b0971656dc69d9aa7646fdcdc58de73795e1b/tests/framework/extensions/clusters/clusters.go#L1096 rather than using a v3 call for a v1 cluster